core/src/main/java/org/elasticsearch/action/index/IndexRequest.java

```diff
@@ -606,7 +615,11 @@ public void writeTo(StreamOutput out) throws IOException {
     public String toString() {
         String sSource = "_na_";
         try {
-            sSource = XContentHelper.convertToJson(source, false);
+            if (source.length() > MAX_SOURCE_LENGTH_IN_TOSTRING) {
+                sSource = "too big: " + ByteSizeValue.formatBytesSizeValue(source.length());
```

Comment:
nik9000 commented above in favor of showing the actual source length ("I like including the original size here.")